hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/utils/FrameUtils.java

```diff
@@ -136,11 +136,15 @@ public static boolean unqualifiedDelegateDetected(final MessageFrame frame) {
         if (!isDelegateCall(frame)) {
             return false;
         }
+        final Address recipient = frame.getRecipientAddress();
 
-        final var recipient = frame.getRecipientAddress();
-        // but we accept delegates if the token redirect contract calls us,
-        // so if they are not a token, then we are a delegate and we are done.
-        if (isToken(frame, recipient)) {
+        final var contractsConfig = configOf(frame).getConfigData(ContractsConfig.class);
+
+        // Evaluate whether the recipient is either a token or on the permitted callers list.
+        // This determines if we should treat this as a delegate call.
+        // We accept delegates if the token redirect contract calls us.
+        if (isToken(frame, recipient)
+                || contractsConfig.permittedDelegateCallers().contains(recipient)) {
```

Comment:
Per @netopyr's point, because the current mono-service value of this property here is **never** going to change, I don't see an issue with making `contractsConfig.permittedDelegateCallers()` return a `Set<Long>` and then doing,
```
if (isToken(frame, recipient) || (ConversionUtils.isLongZero(recipient) && 
    contractsConfig.permittedDelegateCallers().contains(
        ConversionUtils.numberOfLongZero(recipient))) {
```

Reply:
Yep, I got it, thank you! I did the implementation with `Set<Long> `